astroquery/wfau/core.py

```diff
@@ -0,0 +1,903 @@
+            may also be entered as strings as specified in the
+            `astropy.coordinates` module.
+        waveband  : str
+            The color filter to download. Must be one of  ``'all'``, ``'J'``,
+            ``'H'``, ``'K'``, ``'H2'``, ``'Z'``, ``'Y'``, ``'Br'``].
+        frame_type : str
+            The type of image. Must be one of ``'stack'``, ``'normal'``,
+            ``'interleave'``, ``'deep_stack'``, ``'confidence'``,
+            ``'difference'``, ``'leavstack'``, ``'all'``]
+        image_width : str or `~astropy.units.Quantity` object, optional
+            The image size (along X). Cannot exceed 15 arcmin. If missing,
+            defaults to 1 arcmin.
+        image_height : str or `~astropy.units.Quantity` object, optional
+             The image size (along Y). Cannot exceed 90 arcmin. If missing,
+             same as image_width.
+        radius : str or `~astropy.units.Quantity` object, optional
+            The string must be parsable by `~astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from `astropy.units`
+            may also be used. When missing only image around the given position
+            rather than multi-frames are retrieved.
+        programme_id : str
+            The survey or programme in which to search for.
+        database : str
+            The WFAU database to use.
+        verbose : bool
+            Defaults to `True`. When `True` prints additional messages.
+        get_query_payload : bool, optional
+            If `True` then returns the dictionary sent as the HTTP request.
+            Defaults to `False`.
+
+        Returns
+        -------
+        list : A list of `~astropy.io.fits.HDUList` objects.
+        """
+
+        if database is None:
+            database = self.database
+
+        if programme_id is None:
+            programme_id = self.programme_id
+
+        readable_objs = self.get_images_async(
+            coordinates, waveband=waveband, frame_type=frame_type,
+            image_width=image_width, image_height=image_height,
+            database=database, programme_id=programme_id, radius=radius,
+            verbose=verbose, get_query_payload=get_query_payload,
+            show_progress=show_progress)
+
+        if get_query_payload:
+            return readable_objs
+        return [obj.get_fits() for obj in readable_objs]
+
+    def get_images_async(self, coordinates, waveband='all', frame_type='stack',
+                         image_width=1 * u.arcmin, image_height=None,
+                         radius=None, database=None,
+                         programme_id=None, verbose=True,
+                         get_query_payload=False,
+                         show_progress=True):
+        """
+        Serves the same purpose as
+        `~astroquery.wfau.BaseWFAUClass.get_images` but returns a list of
+        file handlers to remote files.
+
+        Parameters
```

Comment:
you could use utils.prepend_doctring here to avoid duplicating lines. Actually probably we could include that in ``async_to_sync`` in a separate PR (if you agree, please open an issue).

Reply:
yes, separate PR

Reply:
https://github.com/astropy/astroquery/issues/1030